packages/care-package/import-vapp/src/BuildActions.ts

```diff
@@ -0,0 +1,116 @@
+import * as careDef from '@vcd/care-package-def';
+import { JSONSchema7 } from 'json-schema';
+import {LegacyApiClient} from '@vcd/node-client/lib/legacy.api.client';
+import {Configuration} from './Configuration';
+import {
+    AdminCatalogType, CatalogItemType, InstantiateVAppTemplateParamsType,
+    QueryResultRecordsType,
+    UploadVAppTemplateParamsType
+} from '@vcd/bindings/vcloud/api/rest/schema_v1_5';
+import {CloudDirectorConfig} from '@vcd/node-client';
+
+export class BuildActions implements careDef.BuildActions {
+
+    legacyApiClient: LegacyApiClient;
+
+    getInputSchema(action: string): JSONSchema7 {
+        if (action === 'generate') {
+            return {
+                type: 'object',
+                properties: {
+                    name: {
+                        type: 'string',
+                        description: 'element name'
+                    }
+                }
+            };
+        }
+        return null;
+    }
+
+    getConfiguration?(): any {
+        return {
+            catalog: 'care-package',
+            vAppTemplateName: '',
+            vAppTemplateURL: ''
+        };
+    }
+
+    private async executeRequests(element: careDef.ElementSource) {
+        console.log('executeRequests');
+
+        const configuration: Configuration = element.configuration;
+        console.log('configuration: ' +  JSON.stringify(configuration));
+
+        const catalogId = await this.getOrCreateCatalogId(configuration);
+
+        await this.uploadVAppTemplate(catalogId, configuration);
+    }
+
+    private async getOrCreateCatalogId(configuration: Configuration) {
+        const catalogQuery = '/api/query?type=catalog&filter=name==' + configuration.catalog;
+        console.log('catalog_query: ' + catalogQuery);
+
+        const catalogs = await this.legacyApiClient.get<QueryResultRecordsType>(catalogQuery);
+        console.log(catalogs);
+
+        if (catalogs.total === 0) {
+            console.log('No care package catalog available. Creating...');
+            const adminCatalogType: AdminCatalogType = {
+                name: configuration.catalog,
+                description: 'Catalog created by the care package.',
+                isPublished: true
+            };
+
+            const catalogPost = '/api/admin/org/87ae2b63-da4e-4602-af80-0f722e1a1394/catalogs';
```

Comment:
Should this ID be a configuration?

Reply:
yes, this is draft version, intentionally left it hard coded as setting it properly will be entirely different PR